pinot-common/src/main/java/com/linkedin/pinot/common/metadata/segment/SegmentZKMetadata.java

```diff
@@ -248,6 +259,11 @@ public ZNRecord toZNRecord() {
     ZNRecord znRecord = new ZNRecord(_segmentName);
     znRecord.setSimpleField(CommonConstants.Segment.SEGMENT_NAME, _segmentName);
     znRecord.setSimpleField(CommonConstants.Segment.TABLE_NAME, _tableName);
+
+    if (_crypterName != null) {
```

Comment:
Do we want to change this to set the default crypter if not otherwise set?

Reply:
I debated about it, not sure if we want to pay the price of copy done by DefaultPinotCrypter.